vggish_audio_segmenter.py

```diff
@@ -1,31 +1,116 @@
 
     :param embedding_dim: the output dimensionality of the embedding
     """
 
-    def __init__(self, embedding_dim: int = 128, *args, **kwargs ):
+    def __init__(self, sampling_factor: int=2, chunk_duration: int = 10,
+                sample_rate: int = 44100, default_traversal_paths: Iterable[str] = ['r'], *args, **kwargs ):
         super().__init__(*args, **kwargs)
-        self._dim = embedding_dim
+        self.sampling_factor = sampling_factor # i.e. the n in sampling notation s(nT)
+        self.chunk_duration = chunk_duration
+        self.sample_rate = sample_rate
+        self.default_traversal_paths = default_traversal_paths
+
 
     @requests
-    def segment(self, docs: Optional[DocumentArray], **kwargs):
+    def segment(self, docs: Optional[DocumentArray], parameters: dict, **kwargs):
         """
-        Encode all docs with images and store the encodings in the embedding attribute of the docs.
-        :param docs: documents sent to the encoder. The docs must have `blob` of the shape `256`.
+        Encode all docs with audio and store the segmented regions in the chunks attribute of the docs.
+
+        :param docs: documents sent to the segmenter.
         """
         if not docs:
             return
-        for doc in docs:
-            assert doc.blob.shape[0] == 256
-            doc.embedding = np.random.rand(self._dim)
\ No newline at end of file
+
+        filtered_docs = self._get_input_data(docs, parameters)
+
+        for idx, doc in enumerate(filtered_docs):
+            # a chunk consists of samples collected during chunk_duration
+            chunk_size = int(self.chunk_duration * self.sample_rate) # number of samples
+            num_channels = doc.blob.ndim
+            channel_tags = ('mono',) if num_channels == 1 else ('left', 'right')
+
+            '''
+            With sampling factor of 2, sampling rate of 44.1kHz, and doc blob of length 204800, 
+                the beginning and end sequence of chunks should be :
+                
+                beg=0,    end=44100
+                beg=22050,    end=66150
+                beg=44100,    end=88200
+                beg=66150,    end=110250
+                beg=88200,    end=132300
+                beg=110250,    end=154350
+                beg=132300,    end=176400
+                beg=154350,    end=198450
+                beg=176400,    end=220500
+            '''
+
+            if num_channels==1:
+                num_chunks_per_channel = int(doc.blob.shape[0] / chunk_size)*self.sampling_factor + 1
+                for chunk_id in range(num_chunks_per_channel):
+                    beg = int(chunk_id * chunk_size/self.sampling_factor)
+                    end = beg + chunk_size
+                    if end > doc.blob.size:
+                        continue
+                    doc.chunks.append(
+                        Document(
+                            blob=doc.blob[beg:end],
+                            offset=idx,
```

Comment:
I think this logic does not make sense. Why the offset depends on the position of the `doc` in the `request`?

Reply:
not sure what its original intention was, removed for now